saskatoon/harvest/api.py

```diff
@@ -220,3 +224,176 @@ def list(self, request, *args, **kwargs):
                                  'title': _("New Person")
                                  }
                          })
+
+
+class StatsView(LoginRequiredMixin, generics.ListAPIView):
+    template_name = "app/stats.html"
+    queryset = Harvest.objects.filter(status="Succeeded")
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = HarvestFilter
+    filterset_fields = ('status', 'season')
+    permission_classes = [IsCoreOrAdmin]
+
+    def get_queryset(self):
+        self.season = self.request.query_params.get('season')
+        if self.season is not None:
+            return self.queryset.filter(start_date__year=self.season)
+        return super().get_queryset()
```

Comment:
Do you still need this? The `season` filter called by passing `?season=2021` to the url should already take action when calling `self.filter_queryset()` no ? This was my point about not filtering the queryset manually but simply letting Rest handling it since we already have a `season` field in `HarvestFilter`. Let me know if I'm missing something.

Reply:
I tried that yeah. it gave me this error: `'StatsView' object has no attribute 'season'`.
I followed how the [documentation](https://www.django-rest-framework.org/api-guide/filtering/#filtering-against-query-parameters) used filtering against query parameters.